src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingTavern.java

```diff
@@ -99,6 +99,7 @@ public void onPlayerEnterBuilding(final PlayerEntity player)
     @Override
     public void onColonyTick(@NotNull final IColony colony)
     {
+        super.onColonyTick(colony);
         if (musicCooldown > 0)
```

Comment:
really, that was the problem?

Reply:
ye forgot the super call which has the autohiring logic :D